### PR TITLE
Check for MKL, don't use if not found

### PR DESCRIPTION
**Pull request by Dekken**

### CODE REVIEW

**Comment by Dekken** [comment]
seems to be a problem on Travis

File: tools/python/blas/check_mkl.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+from setuptools import find_packages, setup, Command
+from setuptools.command.install import install
+from setuptools.extension import Extension
+
+modules = [
+            Extension("mkl", sources=["tools/python/blas/mkl.cpp"], language="c++",)
+          ]
+
+class BLASBuild(build):
```
Comment (Mbompr) [inline tools/python/blas/check_mkl.py]: MKLBuild

File: tools/python/blas/check_mkl.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    def run(self):
+        self.run_command('build_ext')
+        build.run(self)
+
+setup(name="checkMKL",
+      version='0.5.0.0',
+      ext_modules=modules,
+      install_requires=['scipy',
+                        'numpydoc',
+                        'scikit-learn'],
```
Comment (Mbompr) [inline tools/python/blas/check_mkl.py]: why `install_requires` ?

File: tools/python/blas/check_mkl.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+# -*- coding: utf8 -*-
+
+# python setup.py build_ext --inplace
+
+##
+# This file exists to check if the system
+#  being used to compile Tick has support
+#  for cblas - the "cblas.cpp" file attempts
```
Comment (Mbompr) [inline tools/python/blas/check_mkl.py]: support for mkl ?

File: tools/python/blas/mkn.cpp [hunk 4]
```diff
@@ -0,0 +1,8 @@
+/*
```
Comment (Mbompr) [inline tools/python/blas/mkn.cpp]: Shouldn't this file be called mkl.cpp instead of mkn.cpp ?

File: setup.py [hunk 5]
```diff
@@ -105,6 +105,16 @@
             print("Error executing check_cblas.py - cblas not found")
     elif 'bdist_wheel' not in sys.argv:
         blas_info = get_info("blas_opt", 0)
+    elif platform.system() != 'Linux': # is mac
```
Comment (Mbompr) [inline setup.py]: Why don't we check for mkl on linux ?

File: tools/python/blas/check_mkl.py (hunk 2 above)
Comment (Dekken) [reply tools/python/blas/check_mkl.py]: copy paste

File: tools/python/blas/check_mkl.py (hunk 1 above)
Comment (Dekken) [reply tools/python/blas/check_mkl.py]: copy pasta

File: tools/python/blas/mkn.cpp (hunk 4 above)
Comment (Dekken) [reply tools/python/blas/mkn.cpp]: yeep, this happens more often that I'd care to admit

File: setup.py (hunk 5 above)
Comment (Dekken) [reply setup.py]: this issue is mostly about conda on mac, which comes with MKL, but we fail to find the headers

File: tools/python/blas/check_mkl.py (hunk 3 above)
Comment (Dekken) [reply tools/python/blas/check_mkl.py]: yes

File: setup.py [hunk 6]
```diff
@@ -31,6 +31,11 @@
 from setuptools.command.install import install
 from setuptools.extension import Extension
 
+force_blas = 0
```
Comment (Mbompr) [inline setup.py]: Please use True and False :)

**Pull request merged**